astronomy/date_time.hpp

```diff
@@ -84,27 +76,51 @@ class DateTime final {
 
   constexpr Date const& date() const;
   constexpr Time const& time() const;
+  // Returns true iff the object was constructed from an ISO 8601 string.
+  constexpr bool iso() const;
```

Comment:
Or from `BeginningOfDay`... Really what we care about in UTC is that it's not constructed from JD or MJD, not that it's constructed from an ISO 8601 string. Switch to `jd()` (which also encompasses any [variants](https://en.wikipedia.org/wiki/Julian_day#Variants), including MJD).